### ISSUE TITLE
Update Dark Mode color mapping for 4.0

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#3122]
### Description
As a follow-up from a Teams conversation, we should explore "inverting" the dark mode paradigm to provide a better experience. Foreground colors should be lighter than background colors.

This is pretty standard across competitor design systems and seems like a good improvement. 

Link to Teams meeting where this was discussed: https://teams.microsoft.com/l/message/19:fd15b51dacd24e70895ec1218a54ae06@thread.skype/1626380916800?tenantId=aee6e3c9-711e-4c7c-bd27-04f2307db20d&groupId=56fae21a-9407-4943-859f-a9bfcf0bbad3&parentMessageId=1626380916800&teamName=Calcite%20Design%20System&channelName=Calcite%20Components&createdTime=1626380916800

https://material.io/design/color/dark-theme.html#properties
### Calcite solution
🔗 [Link to Figma specs](https://www.figma.com/design/VFuWrT6subFRlazKMJhpVK/Dark-mode-color-remap?node-id=0-1&node-type=canvas&t=ttGeh4dPTCvNYWYW-0)

Update dark mode colors:

- [x] background=`core.color.neutral.blk-210`
- [x] foreground.2=`core.color.neutral.blk-190`
- [x] foreground.3=`core.color.neutral.blk-180`
- [x] inverse.hover=`core.color.neutral.blk-010`
- [x] inverse.press=`core.color.neutral.blk-020`

![Image](https://github.com/user-attachments/assets/1976ea4a-a57b-48ae-889a-bfe7167ca5b3)

![Image](https://github.com/user-attachments/assets/b64d8760-2e7d-4527-ad88-631c196ac40d)

![Image](https://github.com/user-attachments/assets/f35f9c1d-2ea4-46aa-9623-e22b5ebed5d8)

![Image](https://github.com/user-attachments/assets/8a5aab85-28fc-41fd-a803-61c4de6c2fe3)

### Relevant Info 

Part of 4.0's Phase I - Colors

Would require updating calcite-colors repository with resultant changes.

**monday.com sync:** #9364439958

**Comment by geospatialem** [comment]
This issue is `blocked` awaiting creation of https://github.com/Esri/calcite-components/issues/863, which was included in the design phase of the new component, `calcite-textarea`. 

Prior to addressing this enhancement, the `textarea` component will be developed and implemented. cc @ashetland

**Comment by SkyeSeitz** [comment]
Even while retaining the same box-shadow values as light mode, similar to [Material Design](https://m2.material.io/design/color/dark-theme.html#properties) (note below), this could improve the box-shadow contrast with surfaces in dark mode, which is currently pretty subtle. 
 
 <img width="744" alt="image" src="https://github.com/Esri/calcite-design-system/assets/29716057/983e4928-ac76-45d0-b48e-a00be7eefb71">

**Comment by geospatialem** [comment]
@SkyeSeitz With the above considerations, should we move this issue out of the freezer and consider it for a future design sprint?

**Comment by SkyeSeitz** [comment]
@geospatialem I don't believe there any issues with moving this issue out of the freezer 👍🏽 
cc: @macandcheese for confirmation

**Comment by DintaMel** [comment]
Verified using @4.0.0-next.35 in [codepen](https://codepen.io/DintaMel-the-lessful/pen/VYeByBL)

### Expected
background=core.color.neutral.blk-210 = #212121
 
foreground.2=core.color.neutral.blk-190 = #363636
 
foreground.3=core.color.neutral.blk-180 = #404040
 
inverse.hover=core.color.neutral.blk-010 = #f2f2f2
 
inverse.press=core.color.neutral.blk-020 = #ebebeb


### Verified

**Background**

<img width="1018" height="173" alt="Image" src="https://github.com/user-attachments/assets/7ffb7415-4e05-4cd7-80b6-9523f5757843" />
<br/>
<br/>

**Foreground 2**

<img width="944" height="288" alt="Image" src="https://github.com/user-attachments/assets/b2bdcd1d-d1d1-4065-82b0-a24b6f18e70b" />

<br/>
<br/>

**Foreground 3**

<img width="949" height="251" alt="Image" src="https://github.com/user-attachments/assets/49bf6b9b-8deb-4605-bb03-d1e2048b6363" />

**Inverse Hover**

<img width="947" height="346" alt="Image" src="https://github.com/user-attachments/assets/df71d30f-5d72-4661-a884-63ce6b5163b8" />

<br/>
<br/>

**Inverse Press**

<img width="966" height="331" alt="Image" src="https://github.com/user-attachments/assets/5c4fba44-5b7b-4f5d-bc6f-5fdc6ac77312" />

<br/>
